katacomb/katacomb/img_facade.py

```diff
@@ -471,12 +475,15 @@ def FitMF(self, **kwargs):
 
         # Check we have the right type of image
         if self.isObitImageMF():
-            try:
-                # Need an ImageMF pointer for PFitSpec
-                ImageMF.PFitSpec(self.imgMF, self._err, **kwargs)
-            except Exception:
-                raise Exception(err_msg)
-            handle_obit_err(err_msg)
+            with img_factory(aips_path=out_path, mode='rw') as out_img:
+                try:
+                    # Copy the structure of self.img to out_img
+                    self.img.Clone(out_img.img, self._err)
+                    # Use an imgMF pointer for PFitSpec2
+                    ImageMF.PFitSpec2(self.imgMF, out_img.imgMF, self._err, **kwargs)
+                except Exception:
+                    raise Exception(err_msg)
+                handle_obit_err(err_msg)
```

Comment:
Any reason `self._err` isn't being passed to `handle_obit_err` here?

Reply:
`self._err` comes from the context in which the pipeline is run, in this case an `ObitContext` see: https://github.com/ska-sa/katsdpcontim/blob/master/katacomb/katacomb/obit_context.py#L20 . `handle_obit_error` will default to the error handler defined in the current context if it isn't set (`self._err` is just defined by retrieving the one from the current context and is passed to any Obit task that requires it).